WebService/WebService/Controllers/BarsController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using WebService.Base_Classes;
+using WebService.Data;
+
+namespace WebService.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BarsController : ControllerBase
+    {
+        private readonly DBContext _context;
+
+        public BarsController(DBContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Bars
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Bar>>> GetBar()
+        {
+            return await _context.Bar.ToListAsync();
+        }
```

Comment:
Shouldn't we use DbAgent method here instead of directly getting it from context? The difference is that DbAgent creates a new instance of context and closes it after.